File: [23.12]Design-Pattern/Java/src/structural/proxy/VirtualProxy.java
package structural.proxy;

public class VirtualProxy implements Image {
    String path;


    public VirtualProxy(String path) {
        this.path = path;
        System.out.println("["+path +"] 경로의 프록시 생성");
    }

    @Override
    public void showImage() {
        HighImage highImage = new HighImage(this.path);
        highImage.showImage();
    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/structural/proxy/VirtualProxy.kt
package structural.proxy

class VirtualProxy(val path:String): Image {

    init{
        println("$path 경로 프록시 생성")
    }
    
    override fun showImage() {
        val highImage = HighImage(this.path)
        highImage.showImage()    
    }
}